### ISSUE TITLE
Test Subset

### ISSUE THREAD

**Issue by VityaVitalich** [minhaoJ2/TaxoEnrich#7]
Dear Maintainers,

I tried to figure out how the test set is formed for SemEval datasets and MAG_PSY but did not succseed. For MAG_CS it is already inside the folder, so there is no problem.

However, it is not the same for other. Inside the code for embedding_generation i find pre-defined path only to CS dataset ```'./data/MAG_CS/computer_science.terms'```

Is it possible to provide other test subsets? Just the vertices for the enrichment would be enough already

**Comment by minhaoJ2** [comment]
Please use the embedding generation file and generate_dataset_binary.py to create the train/valid/test partitions.

**Comment by VityaVitalich** [comment]
> Please use the embedding generation file and generate_dataset_binary.py to create the train/valid/test partitions.

Thank you for your reply! However, I got some troubles with installing proper version of DGL and running it on my local machine, that I have reported over there. You said that publishing requirements is complex and everything worked for you on your side. Therefore, it would be great if you could run it and publish the same way it is done with MAG_CS

https://github.com/minhaoJ2/TaxoEnrich/issues/6

**Comment by VityaVitalich** [comment]
> Please use the embedding generation file and generate_dataset_binary.py to create the train/valid/test partitions.

Moreover, the embedding generation py script contains pre-defined path to the MAG_CS

```    with open('./data/MAG_CS/computer_science.terms.bertembed', 'w') as f:
        f.write(f"{len(taxoid2bertemb)} 768\n")
        for ele in sorted(taxoid2bertemb.items(), key=lambda x: x[0]):
            embed_string = " ".join([str(a.item()) for a in ele[1]])
            f.write(f"{ele[0]} {embed_string}\n")

    with open('./data/MAG_CS/computer_science.terms.train', 'w') as f:
        for node in train_node_ids:
            id = taxoord2id[node]
            f.write(f"{id}\t{taxoid2term[id]}\n")

    with open('./data/MAG_CS/computer_science.terms.validation', 'w') as f:
        for node in validation_node_ids:
            id = taxoord2id[node]
            f.write(f"{id}\t{taxoid2term[id]}\n")

    with open('./data/MAG_CS/computer_science.terms.test', 'w') as f:
        for node in test_node_ids:
            id = taxoord2id[node]
            f.write(f"{id}\t{taxoid2term[id]}\n")
```

This code will surely not generate any files for other subsets. Please could you fix this, so the code is reproducible.

**Comment by VityaVitalich** [comment]
As I observe in embedding_generation script, it requires binary pickle file
```with open("./data/MAG_CS/computer_science.pickle.bin", "rb") as f:```

such file is not uploaded for other than MAG_CS datasets

As i see, the binary file could be created with generate_dataset_binary.py, that is somehow 2nd step after embedding_generation. And this script requires already generated embeddings
It looks like cirricular problem, that first embeddings should be created with the script that is not included

**Comment by minhaoJ2** [comment]
You can first generate the train/val/test splits and then use the pickle file generated to read the taxonomy and other things. Generating embeddings only require the data from taxo and terms files for train, test, val splits separately.

**Comment by VityaVitalich** [comment]
That seems to be more complex. 

I need the embeddings to perform split. Embeddings are loaded inside the split code.

Okay, then i follow the script for embedding generation. First line in main function is ``` with open('./data/MAG_CS/computer_science.pickle.bin', 'rb') as f:```
it requires some pickle binary, that is not presented anywhere

It implies that i need to have embedding before, but it is not clear on how to obtain them and which pretrained model to choose.

**Comment by minhaoJ2** [comment]
You will need the other files to generate the splits first. Then generate the embeddings using the BERT and SciBERT models. Then you can store the embeddings in binary files again. The steps are instructed in README.

**Comment by VityaVitalich** [comment]
Does it mean that first portion of embeddings that are needed for split are not used later and could be generated with any pre-trained model?

**Comment by minhaoJ2** [comment]
As long as you have the splits, it would be fine. The embedding files would be the files generated by embedding generation part.

**Comment by VityaVitalich** [comment]
Got it, Thank you!
